opm/output/eclipse/EclipseReader.cpp

```diff
@@ -17,230 +17,163 @@
-
-            for (size_t index = 0; index < simulator_state.temperature().size(); ++index) {
-                simulator_state.temperature()[index] = unit::convert::from((double)temperature_data[index] - offset, scaling);
-            }
-          } else {
-              throw std::runtime_error("Read of restart file: File does not contain TEMP data\n");
-          }
-    }
-
-
-    void restorePressureData(const ecl_file_type* file,
-                             EclipseStateConstPtr eclipse_state,
-                             int numcells,
-                             SimulationDataContainer& simulator_state) {
-        const char* pressure = "PRESSURE";
-
-        if (ecl_file_has_kw(file , pressure)) {
-
-            ecl_kw_type* pressure_kw = ecl_file_iget_named_kw(file, pressure, 0);
+namespace Opm {
+namespace {
 
-            if (ecl_kw_get_size(pressure_kw) != numcells) {
-                throw std::runtime_error("Read of restart file: Could not restore pressure data, length of data from file not equal number of cells");
-            }
+    inline data::Solution restoreSOLUTION( ecl_file_type* file,
+                                           int numcells,
+                                           const UnitSystem& units ) {
 
-            float* pressure_data = ecl_kw_get_float_ptr(pressure_kw);
-            const double deck_pressure_unit = (eclipse_state->getDeckUnitSystem().getType() == UnitSystem::UNIT_TYPE_METRIC) ? Opm::unit::barsa : Opm::unit::psia;
-            for (size_t index = 0; index < simulator_state.pressure().size(); ++index) {
-                simulator_state.pressure()[index] = unit::convert::from((double)pressure_data[index], deck_pressure_unit);
-            }
-        } else {
-            throw std::runtime_error("Read of restart file: File does not contain PRESSURE data\n");
-        }
-    }
-
-
-    static void restoreSaturation(const ecl_file_type* file_type,
-                                  const PhaseUsage& phaseUsage,
-                                  int numcells,
-                                  SimulationDataContainer& simulator_state) {
-
-        float* sgas_data = NULL;
-        float* swat_data = NULL;
-
-        if (phaseUsage.phase_used[BlackoilPhases::Aqua]) {
-            const char* swat = "SWAT";
-            if (ecl_file_has_kw(file_type, swat)) {
-                ecl_kw_type* swat_kw = ecl_file_iget_named_kw(file_type , swat, 0);
-                swat_data = ecl_kw_get_float_ptr(swat_kw);
-                std::vector<double> swat_datavec(&swat_data[0], &swat_data[numcells]);
-                EclipseIOUtil::addToStripedData(swat_datavec, simulator_state.saturation(), phaseUsage.phase_pos[BlackoilPhases::Aqua], phaseUsage.num_phases);
-            } else {
-                std::string error_str = "Restart file is missing SWAT data!\n";
-                throw std::runtime_error(error_str);
-            }
+        for( const auto* key : { "PRESSURE", "TEMP", "SWAT", "SGAS" } ) {
```

Comment:
`PhaseUsage` is unavailable, and this is currently just a port of old behaviour. I've added it on the TODO list.